### PR TITLE
merkaartor: adopt package

### PR DESCRIPTION
**Pull request by sikmir**
###### Motivation for this change
* enable on darwin
* enable translations
* enable geoimage support
* enable gpsd support
* cleanup buildInputs, compile flags

###### Things done
- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 114401` at cc064e35 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li><a href="https://gist.github.com/fb4afab173b7b3708cb3f40aa10ca803">merkaartor</a></li></li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: unused-argument

Unused argument: `withLibproxy`.
Near pkgs/applications/misc/merkaartor/default.nix:15:3:
```
   |
15 | , withLibproxy ? false, libproxy
   |   ^
```
</li>
  </ul>
</details>

File: pkgs/applications/misc/merkaartor/default.nix [hunk 1]
```diff
@@ -12,25 +27,46 @@ mkDerivation rec {
...
-    (fetchpatch {
-      url = "https://github.com/openstreetmap/merkaartor/commit/e72553a7ea2c7ba0634cc3afcd27a9f7cfef089c.patch";
-      sha256 = "NAisplnS3xHSlRpX+fH15NpbaD+uM57OCsTYGKlIR7U=";
-    })
-  ];
+  patches = [(fetchpatch {
+    # Fix build with Qt 5.15 (missing QPainterPath include)
+    url = "https://github.com/openstreetmap/merkaartor/commit/e72553a7ea2c7ba0634cc3afcd27a9f7cfef089c.patch";
+    sha256 = "NAisplnS3xHSlRpX+fH15NpbaD+uM57OCsTYGKlIR7U=";
+  })] ++ lib.optional withGpsdlib (fetchpatch {
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/merkaartor/default.nix:34]:
```suggestion
  patches = [
    # Fix build with Qt 5.15 (missing QPainterPath include)
    (fetchpatch {
      url = "https://github.com/openstreetmap/merkaartor/commit/e72553a7ea2c7ba0634cc3afcd27a9f7cfef089c.patch";
      sha256 = "NAisplnS3xHSlRpX+fH15NpbaD+uM57OCsTYGKlIR7U=";
    })
  ] ++ lib.optionals withGpsdlib [
    (fetchpatch {
```

File: pkgs/applications/misc/merkaartor/default.nix [hunk 2]
```diff
@@ -12,25 +27,46 @@ mkDerivation rec {
...
+    # Added a condition to use the new timespec_t on gpsd APIs >= 9
+    url = "https://github.com/openstreetmap/merkaartor/commit/13b358fa7899bb34e277b32a4c0d92833050f2c6.patch";
+    sha256 = "129fpjm7illz7ngx3shps5ivrxwf14apw55842xhskwwb0rf5szb";
+  });
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/merkaartor/default.nix:38]:
```suggestion
    })
  ];
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/applications/misc/merkaartor/default.nix [hunk 3]
```diff
@@ -13,24 +28,49 @@ mkDerivation rec {
   };
 
   patches = [
+    # Fix build with Qt 5.15 (missing QPainterPath include)
     (fetchpatch {
       url = "https://github.com/openstreetmap/merkaartor/commit/e72553a7ea2c7ba0634cc3afcd27a9f7cfef089c.patch";
       sha256 = "NAisplnS3xHSlRpX+fH15NpbaD+uM57OCsTYGKlIR7U=";
     })
+  ] ++ lib.optionals withGpsdlib [
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/merkaartor/default.nix:36]: Can we apply this unconditionally to prevent bit rotting?

Comment (sikmir) [reply pkgs/applications/misc/merkaartor/default.nix]: Done

**Comment by hjones2199** [comment]
Result of `nixpkgs-review pr 114401` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>merkaartor</li>
  </ul>
</details>

**Pull request merged**